api/external/cfgmgmt/cfgmgmt.proto

```diff
@@ -25,41 +25,81 @@ import "components/automate-grpc/protoc-gen-policy/api/annotations.proto";
+  Adding a filter makes a list of all nodes that meet the filter criteria. 
+  Supports pagination, filtering, and sorting.
+	*/
   rpc GetNodes(cfgmgmt.request.Nodes) returns (google.protobuf.ListValue) {
     option (google.api.http).get = "/cfgmgmt/nodes";
     option (chef.automate.api.policy).resource = "cfgmgmt:nodes";
     option (chef.automate.api.policy).action = "read";
     option (chef.automate.api.iam.policy).resource = "infra:nodes";
     option (chef.automate.api.iam.policy).action = "infra:nodes:list";
   };
+
+  /*
+  GetRuns
+
+  Returns a list of run ids for the provided node id. 
+  Supports pagination.
+  Accepts a `start` parameter to denote start date for the list and a filter of type `status`.
+	*/
   rpc GetRuns(cfgmgmt.request.Runs) returns (google.protobuf.ListValue) {
     option (google.api.http).get = "/cfgmgmt/nodes/{node_id}/runs";
     option (chef.automate.api.policy).resource = "cfgmgmt:nodes:{node_id}:runs";
     option (chef.automate.api.policy).action = "read";
     option (chef.automate.api.iam.policy).resource = "infra:nodes:{node_id}";
     option (chef.automate.api.iam.policy).action = "infra:nodes:list";
   };
+
+  /*
+  GetNodesCounts
+
+  Returns totals for failed, success, and missing infra nodes that have reported into Automate.
+	*/
   rpc GetNodesCounts (cfgmgmt.request.NodesCounts) returns (cfgmgmt.response.NodesCounts) {
     option (google.api.http).get = "/cfgmgmt/stats/node_counts";
     option (chef.automate.api.policy).resource = "cfgmgmt:stats:node_counts";
     option (chef.automate.api.policy).action = "read";
     option (chef.automate.api.iam.policy).resource = "infra:nodes";
     option (chef.automate.api.iam.policy).action = "infra:nodes:list";
   };
+
+  /*
+  GetRunsCounts
+
+  Returns totals for failed and successful runs given a `node_id`.
+	*/
   rpc GetRunsCounts (cfgmgmt.request.RunsCounts) returns (cfgmgmt.response.RunsCounts) {
     option (google.api.http).get = "/cfgmgmt/stats/run_counts";
     option (chef.automate.api.policy).resource = "cfgmgmt:stats:run_counts";
     option (chef.automate.api.policy).action = "read";
     option (chef.automate.api.iam.policy).resource = "infra:nodes";
     option (chef.automate.api.iam.policy).action = "infra:nodes:list";
   };
+
+  /*
+  GetNodeRun
+
+  Returns the infra run report for the provided node id.
+	*/
   rpc GetNodeRun (cfgmgmt.request.NodeRun) returns (cfgmgmt.response.Run) {
     option (google.api.http).get = "/cfgmgmt/nodes/{node_id}/runs/{run_id}";
     option (chef.automate.api.policy).resource = "cfgmgmt:nodes:{node_id}:runs:{run_id}";
     option (chef.automate.api.policy).action = "read";
     option (chef.automate.api.iam.policy).resource = "infra:nodes:{node_id}";
     option (chef.automate.api.iam.policy).action = "infra:nodes:get";
   };
+
+  /*
+  GetSuggestions
+
+  Returns possible filter values given a valid `type` parameter and at least one character for the `text` parameter.
```

Comment:
It is at least two characters for the text parameter or all values are returned. 
https://github.com/chef/automate/blob/82772a8c0a82da983bb9b329fb31716447d66609/components/config-mgmt-service/backend/elastic/suggestions.go#L121

Reply:
👍  thank you!